nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -1191,9 +1196,51 @@ impl DataStore {
             }
         }
 
+        // Load all pending host_phase_1 updates.
+        let mut pending_updates_host_phase_1 = Vec::new();
+        {
+            #[rustfmt::skip]
+            use nexus_db_schema::schema::bp_pending_mgs_update_host_phase_1::dsl;
+
+            let mut paginator = Paginator::new(
+                SQL_BATCH_SIZE,
+                dropshot::PaginationOrder::Ascending,
+            );
+            while let Some(p) = paginator.next() {
+                let batch = paginated(
+                    dsl::bp_pending_mgs_update_host_phase_1,
+                    dsl::hw_baseboard_id,
+                    &p.current_pagparams(),
+                )
+                .filter(dsl::blueprint_id.eq(to_db_typed_uuid(blueprint_id)))
+                .select(BpPendingMgsUpdateHostPhase1::as_select())
+                .load_async(&*conn)
+                .await
+                .map_err(|e| {
+                    public_error_from_diesel(e, ErrorHandler::Server)
+                })?;
+
+                paginator = p.found_batch(&batch, &|d| d.hw_baseboard_id);
+                for row in batch {
+                    pending_updates_host_phase_1.push(row);
+                }
+            }
+        }
+
         // Collect the unique baseboard ids referenced by pending updates.
-        let baseboard_id_ids: BTreeSet<_> =
-            pending_updates_sp.iter().map(|s| s.hw_baseboard_id).collect();
+        let baseboard_id_ids: BTreeSet<_> = pending_updates_sp
+            .iter()
+            .map(|s| s.hw_baseboard_id)
+            .chain(pending_updates_rot.iter().map(|s| s.hw_baseboard_id))
+            .chain(
+                pending_updates_rot_bootloader
+                    .iter()
+                    .map(|s| s.hw_baseboard_id),
+            )
```

Comment:
This is the bugfix; without this change, trying to read back the blueprints containing only RoT or RoT bootloader updates failed with "should never happen" errors because we weren't inserting the baseboards they needed. I also updated the tests to read back the blueprints they insert to catch this.

Reply:
eek! Thanks for the fix!